File: Chapter05/Java/StorageExample1/app/src/main/java/net/zenconsult/mobile/StoreData.java
package net.zenconsult.mobile;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import java.util.Hashtable;

public class StoreData {

    public static boolean storeData(Hashtable<String, Object> data, Context ctx) {
        SharedPreferences prefs = PreferenceManager.getDefaultSharedPreferences(ctx);
        String hostname = (String) data.get("hostname");
        int port = (Integer) data.get("port");
        boolean useSSL = (Boolean) data.get("ssl");
        SharedPreferences.Editor ed = prefs.edit();
        ed.putString("hostname", hostname);
        ed.putInt("port", port);
        ed.putBoolean("ssl", useSSL);
        return ed.commit();
    }

}

File: Chapter05/Kotlin/StorageExample1/app/src/main/java/net/zenconsult/mobile/StoreData.kt
package net.zenconsult.mobile

import android.content.Context
import androidx.preference.PreferenceManager
import java.util.*

class StoreData {

    companion object {

        fun storeData(data: Hashtable<String, Any?>, context: Context): Boolean {
            val prefs = PreferenceManager.getDefaultSharedPreferences(context)
            val hostname = data["hostname"] as String
            val port = data["port"] as Int
            val useSSL = data["ssl"] as Boolean
            val ed = prefs.edit()
            ed.apply {
                putString("hostname", hostname)
                putInt("port", port)
                putBoolean("ssl", useSSL)
            }
            return ed.commit()
        }

    }
}